cdap-ui/app/hydrator/controllers/create/toppanel-ctrl.js

```diff
@@ -294,10 +304,36 @@ class HydratorPlusPlusTopPanelCtrl {
     seconds = seconds < 10 ? '0' + seconds : seconds;
     minutes = minutes < 10 ? '0' + minutes : minutes;
 
+    this.setDisplayDuration(minutes, seconds);
+  }
+
+  setDisplayDuration(minutes, seconds) {
     this.displayDuration = {
-      minutes: minutes,
-      seconds: seconds
-    };
+      minutes: !minutes ? '--' : minutes,
+      seconds: !seconds ? '--' : seconds,
+    }
+  }
+
+  updateTimerLabelAndTitle(res) {
+    // set default
+    if (!res) {
+      this.timerLabel = 'Duration';
+      this.queueStatus = '';
+      return;
+    }
+
+    const { WAITING, ACQUIRED, INIT, RUNNING } = window.CaskCommon.PREVIEW_STATUS;
+    if (res.status === WAITING && res.positionInWaitingQueue > 0) {
+      const runsAheadInQueue = res.positionInWaitingQueue;
+      this.queueStatus = `${runsAheadInQueue} ${runsAheadInQueue === 1? 'run' : 'runs'} ahead in queue`;
+      this.timerLabel = `${runsAheadInQueue} Pending`;
```

Comment:
Is it possible to define timerLabel in enums? (Pending, Running, Duration)

Reply:
I can't add an enum here so I added the timer labels to global-constants.